### ISSUE TITLE
Resolving 'bad object' errors due to rewritten history change in submodules due to Git LFS migration

### ISSUE THREAD

**Issue by Kitiara** [git-lfs/git-lfs#5706]
Hello,

I've created a test case involving converting a submodule to a Git LFS repository and rewriting its history a couple of commits later using the migrate command. However, I ran into a 'bad object' problem afterward.

After beginning to track some files in a repository with Git LFS, I rewrote the Git history using the 

> git lfs migrate import --everything

 command. However, if this repository was previously added as a submodule to another repository, the super-project-pointer SHA that the commits in the main Git repository reference remained unchanged, resulting in 'fatal: bad object' errors.

In summary, when the history of the submodule repository is rewritten, there was no corresponding change in the main repository, leading to 'bad object' issues in main repository commits. Is there a solution to this problem?

**Comment by bk2204** [comment]
Hey,

This isn't surprising to me based on how submodules work.  A submodule in a Git repository consists of an entry in a tree object pointing to an external commit (of mode `160000`), plus an entry in `.gitmodules` that says where to access the repository.  Because the tree entry refers to a specific object ID, if you remove that object in the submodule, Git will complain about being unable to check it out.

We don't have automatic functionality for rewriting superprojects, but you can use the `--object-map` option to make a map of the old and new object IDs and then rewrite the superproject using that map with something like `git filter-repo` or `git filter-branch`.  However, there's no way to avoid rewriting the superproject history here; if you don't, the submodules can't be accessed in the history.

**Comment by Kitiara** [comment]
Thank you for the idea @bk2204. I have created a python script exactly does that and it looks like it is working as intended. Here it is the script and what i did in case someone may need this in the future. 

I got the map exported with `git lfs migrate import --everything --object-map=map` and then run the script below in the super project as follows: `py fixSuperProject.py "./mySubmodule/map" "mySubmodule"`

```
import sys
import subprocess

if len(sys.argv) < 2:
    print("Usage: py fixSuperProject.py <object_map_file> <submodule_name>")
    sys.exit(1)

object_map_file = sys.argv[1]
submodule_name = sys.argv[2]

id_map = {}

try:
    with open(object_map_file, "r") as f:
        for line in f:
            parts = line.strip().split(",")
            old_id, new_id = parts
            id_map[old_id] = new_id
except FileNotFoundError:
    print(f"File not found: {object_map_file}")

command = [
    "git",
    "filter-repo",
    "--commit-callback",
    f"""
    from pprint import pprint
    for change in commit.file_changes:
        if change.mode.decode("utf-8") == "160000" and change.filename.decode("utf-8") == "{submodule_name}":
            new_blob_id = {id_map}.get(change.blob_id.decode("utf-8"))
            if new_blob_id is not None:
                change.blob_id = new_blob_id.encode("utf-8")
    """
]

result = subprocess.run(command, capture_output=True, text=True)

if result.returncode == 0:
    print(result.stdout)
else:
    print(result.stderr)

```
